Match/FeatureSet.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections.Immutable;
+
+namespace Match
+{
+    public enum Generation
```

Comment:
put enums in a class so there are no naming conflicts?

Reply:
Is there something also called `Generation` I should be aware of?

Reply:
probably should have put this on features instead, incase we want to use it to enable run stuff or smth like that